org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/extensions/XMLExtensionRegistryTest.java

```diff
@@ -0,0 +1,83 @@
+package org.eclipse.lemminx.services.extensions;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.function.Predicate;
+
+import org.eclipse.lemminx.XMLAssert.SettingsSaveContext;
+import org.eclipse.lemminx.XMLLanguageServer;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.services.XMLLanguageService;
+import org.eclipse.lemminx.services.extensions.IXMLExtension;
+import org.eclipse.lemminx.services.extensions.XMLExtensionsRegistry;
+import org.eclipse.lemminx.services.extensions.save.ISaveContext;
+import org.eclipse.lsp4j.InitializeParams;
+import org.junit.jupiter.api.Test;
+
+import com.google.gson.JsonObject;
+
+public class XMLExtensionRegistryTest {
+	
+	class RegistryTestExtension implements IXMLExtension {
+		
+		private boolean extensionDoSaveCalled = false;
+		
+		@Override
+		public void start(InitializeParams params, XMLExtensionsRegistry registry) {
+		}
+
+		@Override
+		public void stop(XMLExtensionsRegistry registry) {
+		}
+		
+		@Override
+		public void doSave(ISaveContext context) {
+			extensionDoSaveCalled = true;
+		}
+
+		public boolean isExtensionDoSaveCalled() {
+			return extensionDoSaveCalled;
+		}
+	}
+	
+	private static class SaveFileContext implements ISaveContext {
+		@Override
+		public DOMDocument getDocument(String uri) {
+			return null;
+		}
+
+		@Override
+		public void collectDocumentToValidate(Predicate<DOMDocument> validateDocumentPredicate) {
+		}
+
+		@Override
+		public SaveContextType getType() {
+			return SaveContextType.DOCUMENT;
+		}
+
+		@Override
+		public String getUri() {
+			return null;
+		}
+
+		@Override
+		public Object getSettings() {
+			return null;
+		}
+	}
+
+	@Test
+	public void initialConfiguration() {
+		XMLLanguageService registry = new XMLLanguageService();
```

Comment:
To fix compilation problem, replace with

`		XMLExtensionsRegistry registry = new XMLExtensionsRegistry();`